addons/website_slides/controllers/main.py

```diff
@@ -856,11 +867,33 @@ def slide_set_completed(self, slide_id):
         fetch_res = self._fetch_slide(slide_id)
         if fetch_res.get('error'):
             return fetch_res
-        self._set_completed_slide(fetch_res['slide'])
+
+        slide = fetch_res['slide']
+        if not slide.channel_id.is_member:
```

Comment:
The necessary checks are already done in "_set_completed_slide", seems redundant.